src/execution/riscv64/simulator-riscv64.cc

```diff
@@ -2818,203 +2853,245 @@ void Simulator::DecodeRVR4Type() {
-void Simulator::DecodeRVIType() {
-  switch (instr_.InstructionBits() & kITypeMask) {
-    case RO_JALR: {
-      set_rd(get_pc() + kInstrSize);
-      // Note: No need to shift 2 for JALR's imm12, but set lowest bit to 0.
-      int64_t next_pc = (rs1() + imm12()) & ~reg_t(1);
-      set_pc(next_pc);
-      break;
-    }
-    case RO_LB: {
-      int64_t addr = rs1() + imm12();
-      int8_t val = ReadMem<int8_t>(addr, instr_.instr());
-      set_rd(sext_xlen(val), false);
-      TraceMemRd(addr, val, get_register(rd_reg()));
-      break;
-    }
-    case RO_LH: {
-      int64_t addr = rs1() + imm12();
-      int16_t val = ReadMem<int16_t>(addr, instr_.instr());
-      set_rd(sext_xlen(val), false);
-      TraceMemRd(addr, val, get_register(rd_reg()));
-      break;
-    }
-    case RO_LW: {
-      int64_t addr = rs1() + imm12();
-      int32_t val = ReadMem<int32_t>(addr, instr_.instr());
-      set_rd(sext_xlen(val), false);
-      TraceMemRd(addr, val, get_register(rd_reg()));
-      break;
-    }
-    case RO_LBU: {
-      int64_t addr = rs1() + imm12();
-      uint8_t val = ReadMem<uint8_t>(addr, instr_.instr());
-      set_rd(zext_xlen(val), false);
-      TraceMemRd(addr, val, get_register(rd_reg()));
-      break;
-    }
-    case RO_LHU: {
-      int64_t addr = rs1() + imm12();
-      uint16_t val = ReadMem<uint16_t>(addr, instr_.instr());
-      set_rd(zext_xlen(val), false);
-      TraceMemRd(addr, val, get_register(rd_reg()));
-      break;
+void Simulator::DecodeRvvVL() {
+  uint32_t instr_temp =
+      instr_.InstructionBits() & (kRvvMopMask | kRvvNfMask | kBaseOpcodeMask);
+  if (instr_temp == RO_V_VL) {
+    reg_t nf = ((instr_.InstructionBits() >> 29) & (kRvvNfMask)) + 1;
+    reg_t vl = rvv_vl();
+    reg_t addr = rs1();
+    reg_t vd = rvv_vd_reg();
+    uint32_t vsew = rvv_vsew();
+
+    // check
+    VI_CHECK_LOAD(vsew);
+    for (reg_t i = 0; i < vl; ++i) {
+      VI_ELEMENT_SKIP(i);
+      set_rvv_vstart(i);
+      for (uint64_t fn = 0; fn < nf; ++fn) {
+        if (vsew == E8) {
+          auto val =
+              ReadMem<type_sew_t<8>::type>(addr + i * nf + fn, instr_.instr());
+          Rvvelt<type_sew_t<8>::type>(vd + fn * emul, i, true) = val;
+        } else if (vsew == E16) {
+          auto val =
+              ReadMem<type_sew_t<16>::type>(addr + i * nf + fn, instr_.instr());
+          Rvvelt<type_sew_t<16>::type>(vd + fn * emul, i, true) = val;
+        } else if (vsew == E32) {
+          auto val =
+              ReadMem<type_sew_t<32>::type>(addr + i * nf + fn, instr_.instr());
+          Rvvelt<type_sew_t<32>::type>(vd + fn * emul, i, true) = val;
+        } else if (vsew == E64) {
+          auto val =
+              ReadMem<type_sew_t<64>::type>(addr + i * nf + fn, instr_.instr());
+          Rvvelt<type_sew_t<64>::type>(vd + fn * emul, i, true) = val;
+        }
+      }
     }
+  }
+  set_rvv_vstart(0);
+}
+void Simulator::DecodeRVIType() {
+  if (switch_sew() != -1) {
+    DecodeRvvVL();
+  } else {
```

Comment:
Alter it. like 
```
if (switch_sew() != -1) {
    DecodeRvvVL();
    return;
  }
switch (instr_.InstructionBits() & kITypeMask) {
```
so it can alter fewer  code